x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorIndexService.java

```diff
@@ -90,11 +99,26 @@ public void createConnectorWithDocId(PutConnectorAction.Request request, ActionL
         );
 
         try {
-            final IndexRequest indexRequest = new IndexRequest(CONNECTOR_INDEX_NAME).opType(DocWriteRequest.OpType.INDEX)
-                .id(request.getConnectorId())
-                .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
-                .source(connector.toXContent(jsonBuilder(), ToXContent.EMPTY_PARAMS));
-            clientWithOrigin.index(indexRequest, listener);
+            isDataIndexNameAlreadyInUse(indexName, connectorId, listener.delegateFailure((l, isIndexNameInUse) -> {
+                if (isIndexNameInUse) {
+                    l.onFailure(
+                        new ElasticsearchStatusException(
+                            "Index name [" + indexName + "] is used by another connector.",
```

Comment:
It could be helpful, if we could add the id of the other connector in the error message. Would make debugging a lot easier for a user.

Reply:
Agreed! I will extends this by using listConnectorByIndexName (that yet needs to be implemented as it's expected by Kibana). I will take care of it in another PR. I don't want to overcomplicate the `isDataIndexNameAlreadyInUse` - it should just return bool imo.